### PR TITLE
[Swift bindings] Add bindings experimental NuGet to the dotnet-experimental feed

### PR DESCRIPTION
**Pull request by kotlarmilos**
## Description

This PR adds binding generation step to the distribution pipeline. To enable testing of the StoreKit framework, we want to provide an experimental NuGet that can be used in real-world applications such as HikingApp. This approach will also allow us to track binding coverage and identify areas for improvement. 

### Changes

This change introduces a script for generating bindings that runs after the build in the distribution pipeline. The script extracts the ABI for selected frameworks, runs the tooling, and generates NuGet package from source code. The package is then distributed using `darc` in the `dotnet-experimental` feed.  

### Verification

The pipeline was invoked manually. The NuGet package will become available once this change is merged into the `feature/swift-bindings` branch, which is configured for feed publishing.

### Follow-up

Once the package becomes available, we will start introducing framework tests as MacCatalyst apps and track coverage of supported Swift language features from included frameworks.

### CODE REVIEW

File: src/Swift.Bindings/tests/FrameworkTests/HikingApp/HikingApp.MacCatalyst/HikingApp.MacCatalyst.csproj [hunk 1]
```diff
@@ -24,9 +24,14 @@
     </ImageAsset>
     <ImageAsset Include="Assets.xcassets\TrailImage.imageset\Contents.json">
       <Visible>false</Visible>
-    </ImageAsset>   
+    </ImageAsset>
   </ItemGroup>
 
+  <!-- Reference Swift.Bindings.MacOSX.Experimental from dotnet-experimental feed -->
```
Comment (jkotas) [inline src/Swift.Bindings/tests/FrameworkTests/HikingApp/HikingApp.MacCatalyst/HikingApp.MacCatalyst.csproj]: For tests, I think you want to reference local copy of the package that was just built.

**Comment by kotlarmilos** [comment]
@EgorBot -codesafety

File: generate.sh [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+cd "$output_dir"
+
+# Function to extract ABI file
+function ExtractABI {
+    local framework=$1
+
+    echo "Generating ABI for framework '$framework', platform '$platform', architecture '$arch'"
+    
+    local sdk_path=$(xcrun -sdk $(echo "$platform" | tr '[:upper:]' '[:lower:]') --show-sdk-path)
+    local swift_interface_path="/Applications/Xcode.app/Contents/Developer/Platforms/${platform}.platform/Developer/SDKs/${platform}.sdk/System/Library/Frameworks/${framework}.framework/Versions/Current/Modules/${framework}.swiftmodule/${arch}.swiftinterface"
```
Comment (akoeplinger) [inline generate.sh]: nit: use `xcode-select -p` to figure out path to Xcode rather than hardcoding /Applications

File: generate.sh [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+        -module-name "$framework" \
+        -sdk "$sdk_path" \
+        -emit-abi-descriptor-path "./${framework}.abi.json"
+}
+
+# Function to generate bindings
+function InvokeProjectionTooling {
+    local framework=$1
+
+    $scriptroot/.dotnet/dotnet $scriptroot/artifacts/bin/Swift.Bindings/Release/net9.0/Swift.Bindings.dll -a "$framework" -o "./"
```
Comment (akoeplinger) [inline generate.sh]:
```suggestion
    $scriptroot/dotnet.sh $scriptroot/artifacts/bin/Swift.Bindings/Release/net9.0/Swift.Bindings.dll -a "$framework" -o "./"
```

File: generate.sh [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+    <Nullable>enable</Nullable>
+    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+    <IsPackable>true</IsPackable>
+    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
+    <IsAotCompatible>true</IsAotCompatible>
+  </PropertyGroup>
+</Project>
+EOL
+
+    $scriptroot/.dotnet/dotnet pack "$project_file"
```
Comment (akoeplinger) [inline generate.sh]:
```suggestion
    $scriptroot/dotnet.sh pack "$project_file"
```

**Review by akoeplinger — APPROVED** [review]

**Comment by jkurdek** [comment]
This change seems to have broken the vscode build. 

Steps to repro. 
1. Clean artifacts / clone new repo
2. Run build from solution explorer view

**Comment by kotlarmilos** [comment]
Since `dotnet.sh` uses the local version of dotnet, it is necessary to run `./build.sh` initially to download it.

**Pull request merged**